app/views/question_sets/_form.html.erb

```diff
@@ -26,7 +26,62 @@
     <% end %>
   </div>
 
+  <!-- check if current_user's courses have any questions
+        and allow user to filter those questions by searching tags.-->
+  <% if @questions.present? %>
+    <!-- add "search bar" for tags and filter questions shown based on
+          tag entered -->
+
+    <input type="text" id="tagSearchInput" onkeyup="qSetQTagSearch()" placeholder="Search for question tags...">
+
+    <table id="questionTagTable" border="1">
+      <tr>
+        <th></th>
+        <th style="min-width:250px">Body</th>
+        <th style="min-width:100px">Tags</th>
+      </tr>
+      <!-- loop thru questions from controller
+            (questions associated with courses the current_user is
+            a professor or assistant for) -->
+      <% @questions.each do |q| %>
```

Comment:
I have removed the duplicate table.

I tried saving a question set with nothing checked on the current master branch, and it errored out as well. That issue is not related to these changes.